docs/applications/nublado/major-upgrade.rst

```diff
@@ -0,0 +1,313 @@
+   username and the database name are ``nublado3``.  In general you
+   should be consuming a database as a service, in which case you will
+   need to request this database from your DBAs and ask them the
+   password.  If you are instead running the internal database, you will
+   need to create this database and grant permissions in the
+   ``postgres`` application configuration.
+#. ``slack_webhook``: the Slack webhook allowing the JupyterLab
+   controller to post status messages to Slack.  This is the same one as
+   used by, e.g. Gafaelfawr in your RSP instance.
+
+Note that in future the secret setup and management will all be handled
+as part of Phalanx deployment, and you will no longer need to do this
+manually.  However, at the present, you still do.
+
+Enable the application
+""""""""""""""""""""""
+
+In the values file for your installation in the top-level Phalanx
+``environments`` directory, set ``nublado: true``.
+
+Configure nublado v3
+""""""""""""""""""""
+
+In the values file for your installation in ``applications/nublado``,
+set up your site-specific values.  The bulk of this work will be
+specifying the set of mount points you use for persistent user data.
+More than likely, you will use the same mount points for nublado v3 as
+you did for nublado v2, but the configuration syntax differs.
+
+Note that if you do share persistent storage between nublado versions,
+the step about communicating with your users (below) goes from
+"important" to "critical".
+
+
+Create an alternate route
+"""""""""""""""""""""""""
+
+In the values file for your installation in ``applications/nublado``,
+set the key ``jupyterhub.hub.baseUrl`` to ``/n3``.
+
+Communicate to user base
+""""""""""""""""""""""""
+
+Let your users know about the new route and when it will be enabled
+(presumably, at the end of the next maintenance window).  Announce that
+that maintenance window will be disruptive and will destroy any existing
+user sessions.
+
+If you are using the same storage for users regardless of which spawner
+they're coming in as, which you very likely are, in addition to the new
+route, you need to do your utmost to make it extremely clear that if
+they are testing the new route, it is vital to make sure that their labs
+are shut down before changing from one route to the other.  If they do
+not do this, it will at best create confusion as open notebooks will be
+saved every five minutes by each environment, thus bafflingly reverting
+changes, and at worst it will cause full-scale corruption of the files
+the user has open.
+
+
+Silence Mobu complaints
+"""""""""""""""""""""""
+
+Turn off ``mobu`` if you're using it, so that it doesn't complain
+about the service outage.
+
+Disable logins
+""""""""""""""
+
+Disable user logins to nublado v2.  This is most easily done by deleting
+the ``hub`` deployment from nublado v2.
+
+Remove user sessions
+""""""""""""""""""""
+
+Remove any user sessions.  This is not, strictly speaking, necessary,
```

Comment:
I'd just say delete their sessions!  It seems like with the shared homedirs that it kind of is necessary, and either way it's probably safer to delete their session?  You already said above to tell them to turn it off